freqtrade/persistence/models.py

```diff
@@ -856,6 +856,32 @@ def get_overall_performance() -> List[Dict[str, Any]]:
             for pair, profit, profit_abs, count in pair_rates
         ]
 
+    @staticmethod
+    def get_performance(days: int) -> List[Dict[str, Any]]:
+        """
+        Returns List of dicts containing all Trades, including profit and trade count
+        NOTE: Not supported in Backtesting.
+        """
+        start_date = datetime.today() - timedelta(days)
+        pair_rates = Trade.query.with_entities(
+            Trade.pair,
+            func.sum(Trade.close_profit).label('profit_sum'),
+            func.sum(Trade.close_profit_abs).label('profit_sum_abs'),
+            func.count(Trade.pair).label('count')
+        ).filter(Trade.is_open.is_(False) & (Trade.close_date >= start_date))\
+            .group_by(Trade.pair) \
+            .order_by(desc('profit_sum_abs')) \
+            .all()
+        return [
+            {
+                'pair': pair,
+                'profit': profit,
+                'profit_abs': profit_abs,
+                'count': count
+            }
+            for pair, profit, profit_abs, count in pair_rates
```

Comment:
I would think that a better way to do this is by introducing an (optional) argument to the existing method - which (if not None) adds a filter.

the `.filter()` method also accepts an (exploded) list - so you should be able to use something like the below:

``` python
filters = [Trade.is_open.is_(False)]
if days:
    start_date = datetime.today() - timedelta(days)
    filters.append((Trade.close_date >= start_date))

#....

.filter(*filters)\. <...>
```

Reply:
Updated according to your suggestion